src/main.c

```diff
@@ -366,223 +363,172 @@ int main(int argc, char* argv[]) {
     } else if (!strcmp(argv[i], "--")) {
       args_done = 1;
     } else {
-      if (argv[i][1] == 'L') {
-        if (jv_get_kind(lib_search_paths) == JV_KIND_NULL)
-          lib_search_paths = jv_array();
-        if (argv[i][2] != 0) { // -Lname (faster check than strlen)
-            lib_search_paths = jv_array_append(lib_search_paths, jq_realpath(jv_string(argv[i]+2)));
-        } else if (i >= argc - 1) {
-          fprintf(stderr, "-L takes a parameter: (e.g. -L /search/path or -L/search/path)\n");
-          die();
-        } else {
-          lib_search_paths = jv_array_append(lib_search_paths, jq_realpath(jv_string(argv[i+1])));
-          i++;
-        }
-        continue;
-      }
-
-      if (isoption(argv[i], 's', "slurp", &short_opts)) {
-        options |= SLURP;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'r', "raw-output", &short_opts)) {
-        options |= RAW_OUTPUT;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 0, "raw-output0", &short_opts)) {
-        options |= RAW_OUTPUT | RAW_NO_LF | RAW_OUTPUT0;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'j', "join-output", &short_opts)) {
-        options |= RAW_OUTPUT | RAW_NO_LF;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'c', "compact-output", &short_opts)) {
-        dumpopts &= ~(JV_PRINT_TAB | JV_PRINT_INDENT_FLAGS(7));
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'C', "color-output", &short_opts)) {
-        options |= COLOR_OUTPUT;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'M', "monochrome-output", &short_opts)) {
-        options |= NO_COLOR_OUTPUT;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'a', "ascii-output", &short_opts)) {
-        options |= ASCII_OUTPUT;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 0, "unbuffered", &short_opts)) {
-        options |= UNBUFFERED_OUTPUT;
-        continue;
-      }
-      if (isoption(argv[i], 'S', "sort-keys", &short_opts)) {
-        options |= SORTED_OUTPUT;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'R', "raw-input", &short_opts)) {
-        options |= RAW_INPUT;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'n', "null-input", &short_opts)) {
-        options |= PROVIDE_NULL;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'f', "from-file", &short_opts)) {
-        options |= FROM_FILE;
-        if (!short_opts) continue;
-      }
-      if (isoption(argv[i], 'b', "binary", &short_opts)) {
+      const char* text = argv[i];
+      int is_short;
+      // First '-' already checked by isoptish
+      if (text[1] == '-') {
+        text += 2;
+        is_short = 0;
+      } else {
+        text++;
+        is_short = 1;
+      }
+      int raw; // Temporary for --rawfile
```

Comment:
Unfortunately, it can't be made more local and keep the if-else chain. The chain removes the need for all the conditional continues, so I prefer an extra variable over that.

Honestly, I don't think `is_rawfile` gives any extra clarity locally 🤷‍♀️.

Reply:
Ok! work for me, have no strong opinion about it 👍